cibuild.cmd

```diff
@@ -86,7 +86,22 @@ if defined TestPerfCorrectness (
 
 if defined TestPerfRun (
     msbuild %MSBuildAdditionalCommandLineArgs% Roslyn.sln /p:Configuration=%BuildConfiguration% /p:DeployExtension=false || goto :BuildFailed
-    .\Binaries\%BuildConfiguration%\Roslyn.Test.Performance.Runner.exe --no-trace-upload || goto :BuildFailed
+
+    if defined GIT_BRANCH (
+        REM Check if we have credentials to upload to benchview
+        if defined BV_UPLOAD_SAS_TOKEN (
+            set "EXTRA_PERF_RUNNER_ARGS=--report-benchview --branch "%GIT_BRANCH%""
+
+            REM Check if we are in a PR or this is a rolling submission
+            if defined ghprbPullTitle (
+                set "EXTRA_PERF_RUNNER_ARGS=!EXTRA_PERF_RUNNER_ARGS! --benchview-submission-name "[%ghprbPullAuthorLogin%] PR %ghprbPullId%: %ghprbPullTitle%" --benchview-submission-type private"
+            ) else (
+                set "EXTRA_PERF_RUNNER_ARGS=!EXTRA_PERF_RUNNER_ARGS! --benchview-submission-type rolling"
+            )
+        )
```

Comment:
The reason I have this logic here is because the environment variables it utilizes are specific to Jenkins, so it only makes sense in the CI context. I'm happy to pull this out of cibuild.cmd into either its own ciperf.cmd or Perf.Runner.exe, if that is what you'd prefer. I could use some guidance here.